tests-web/www/js/controls.js

```diff
@@ -0,0 +1,197 @@
+// TODO A Leaflet control isn't the right abstraction. We want a popup modal
+// that totally blocks the map and other things. Things like disabling
+// settingsButton to stop multiple of these controls is a total hack.
+const SettingsControl = L.Control.extend({
+  // TODO Centered would be great. https://github.com/Leaflet/Leaflet/issues/8358
+  options: {
+    position: "topleft",
+  },
+  onAdd: function (map) {
+    const checkbox1 = makeCheckbox(
+      "debugEachStep",
+      "Debug each transformation step\n",
+      this.options.app.importSettings.debugEachStep,
+      (checked) => {
+        this.options.app.importSettings.debugEachStep = checked;
+      }
+    );
+    const checkbox2 = makeCheckbox(
+      "dualCarriagewayExperiment",
+      "Enable dual carriageway experiment\n",
+      this.options.app.importSettings.dualCarriagewayExperiment,
+      (checked) => {
+        this.options.app.importSettings.dualCarriagewayExperiment = checked;
+      }
+    );
+
+    const button = L.DomUtil.create("button");
+    button.type = "button";
+    button.innerHTML = "Confirm";
+    button.onclick = () => {
+      this.remove();
+      document.getElementById("settingsButton").disabled = false;
+    };
+
+    var group = makeDiv([checkbox1, checkbox2, button]);
+    group.style = "background: black; padding: 10px;";
+    L.DomEvent.disableClickPropagation(group);
+    return group;
+  },
+});
+
+export const makeSettingsControl = function (app) {
+  return new SettingsControl({ app: app });
+};
+
+export class LayerGroup {
+  constructor(name, map) {
+    this.name = name;
+    this.layers = [];
+    this.enabled = true;
+    this.map = map;
+  }
+
+  addLayer(name, layer, { enabled = true } = {}) {
+    this.layers.push({ name, enabled, layer });
+  }
+
+  // Doesn't re-render
+  setEnabled(enabled) {
+    this.enabled = enabled;
+    for (const layer of this.layers) {
+      layer.enabled = enabled;
+    }
+  }
+
+  renderControls() {
+    var members = [
+      makeCheckbox(
+        this.name,
+        `<u>${this.name}</u>`,
+        this.enabled,
+        (checked) => {
+          this.setEnabled(checked);
+
+          // Rerender
+          document.getElementById(this.name).replaceWith(this.renderControls());
+        }
+      ),
+    ];
+    for (const layer of this.layers) {
+      members.push(
+        makeCheckbox(
+          this.name + "_" + layer.name,
+          layer.name,
+          layer.enabled,
+          (checked) => {
+            layer.enabled = checked;
+            if (checked) {
```

Comment:
Turns out `map.addLayer` and `removeLayer` are idempotent, so we can be a bit sloppy here and just set the state to whatever want.